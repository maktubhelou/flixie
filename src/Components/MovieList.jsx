import React, { Component } from 'react'
import PropTypes from 'prop-types'
import MovieCard from './MovieCard'

const posterBaseUrl = 'https://image.tmdb.org/t/p/'
const posterSize = 'w300'
const fullResUrl = `${posterBaseUrl}original`

const styles = {
  wrapper: {
    minHeight: '300px',
    maxWidth: '300px',
    marginTop: '50px',
  },
  header: {
    color: 'white',
  },
}

class MovieList extends Component {
  render() {
    const { movies } = this.props
    if (movies.length > 0) {
      const elements = movies.map(movie => {
        return (
          <MovieCard
            key={movie.id}
            id={movie.id}
            title={movie.title}
            score={movie.vote_average}
            posterPath={posterBaseUrl + posterSize + movie.poster_path}
            fullResUrl={fullResUrl}
            posterSrc={movie.poster_path}
            releaseDate={movie.release_date}
            overview={movie.overview}
            voteCount={movie.vote_count}
            popularity={Math.floor(movie.popularity)}
            renderLightBox={this.renderLightBox}
          />
        )
      })
      return elements
    }
    return (
      <div style={styles.wrapper}>
        <h1 style={styles.header}>
          There are no movies matching your search! Run away!
        </h1>
      </div>
    )
  }
}

MovieList.propTypes = {
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      vote_average: PropTypes.number,
      poster_path: PropTypes.string,
      release_date: PropTypes.string,
      overview: PropTypes.string,
      vote_count: PropTypes.number,
      popularity: PropTypes.number,
    }),
  ),
}

MovieList.defaultProps = {
  movies: [],
}

export default MovieList
