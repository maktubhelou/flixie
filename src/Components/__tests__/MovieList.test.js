import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MovieList from '../MovieList'

describe('<MovieList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<MovieList />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<MovieList />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
