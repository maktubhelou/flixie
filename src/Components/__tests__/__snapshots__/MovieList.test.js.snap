// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieList /> matches snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": "50px",
      "maxWidth": "300px",
      "minHeight": "300px",
    }
  }
>
  <h1
    style={
      Object {
        "color": "white",
      }
    }
  >
    There are no movies matching your search! Run away!
  </h1>
</div>
`;
